src/main/java/org/testng/internal/TestListenerHelper.java

```diff
@@ -115,26 +117,31 @@ public static ListenerHolder findAllListeners(Class<?> cls, IAnnotationFinder fi
     result.listenerClasses = Lists.newArrayList();
 
     while (cls != Object.class) {
+      List<IListenersAnnotation> annotations = finder.findAnnotations(cls, IListenersAnnotation.class);
       IListenersAnnotation l = finder.findAnnotation(cls, IListenersAnnotation.class);
-      if (l != null) {
-        Class<? extends ITestNGListener>[] classes = l.getValue();
-        for (Class<? extends ITestNGListener> c : classes) {
-          result.listenerClasses.add(c);
-
-          if (ITestNGListenerFactory.class.isAssignableFrom(c)) {
-            if (result.listenerFactoryClass == null) {
-              result.listenerFactoryClass = (Class<? extends ITestNGListenerFactory>) c;
-            } else {
-              throw new TestNGException(
-                  "Found more than one class implementing "
-                      + "ITestNGListenerFactory:"
-                      + c
-                      + " and "
-                      + result.listenerFactoryClass);
+      annotations.add(l);
```

Comment:
`l` is already supposed to be into the list: `findAnnotations(cls, IListenersAnnotation.class)` and `finder.findAnnotation(cls, IListenersAnnotation.class)` should return the same result, except `findAnnotations` will dig in the inheritance.

Reply:
I have changed the findAnnotations to findInheritedAnnotations to ignore misunderstanding

Reply:
In fact, I don't know what is the best behavior. What could be the use case where we don't want to have the annotation from the target class?

@krmahadevan WDYT?

Reply:
@juherr - When you say `target class` am guessing that you are referring to the test class. A user may want to add listeners into base classes (or) interfaces for that matter to just do away with the duplication.

Personally speaking, I dont see a lot of value with the `@Listeners` annotation, because it's not local in nature. Listeners specified via this annotation, get hooked up across all classes, instead of them being locally applicable only to the class (or) class hierarchy that they were found in.